himl/main.py

```diff
@@ -21,6 +21,22 @@ def run(self, args):
         opts = parser.parse_args(args)
         self.do_run(opts)
 
+    """
+    example of custom merge strategy. pass this function to ConfigProcessor to override default merging behavior, which is append
+    """
+    def strategy_merge_override(self,config, path, base, nxt):
+        """merge list of items. if items have same id, nxt replaces base."""
+        """if remove flag is present in nxt item, remove base and not add nxt"""
+        result = deepcopy(base)
+        for nxto in nxt:
+            for baseo in result:
+                if 'id' in baseo and 'id' in nxto and baseo['id'] == nxto['id']:
```

Comment:
What is the list items are nort dicts? But say plain string. Should we add a check for that?

Reply:
done. added type check to avoid this edge case:
 ```
           # if list is not a list of dicts, bail out and let the next strategy to execute
            if not isinstance(baseo,dict) or not isinstance(nxto,dict):
                return STRATEGY_END
```